src/main/java/uk/ac/ebi/ampt2d/storage/FileSystemStorageService.java

```diff
@@ -0,0 +1,82 @@
+package uk.ac.ebi.ampt2d.storage;
+
+import org.apache.log4j.Logger;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.core.io.Resource;
+import org.springframework.core.io.UrlResource;
+import org.springframework.stereotype.Service;
+import org.springframework.util.FileSystemUtils;
+import org.springframework.web.multipart.MultipartFile;
+
+import java.io.IOException;
+import java.net.MalformedURLException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+
+@Service
+public class FileSystemStorageService implements StorageService {
+    private Logger logger = Logger.getLogger(FileSystemStorageService.class);
+
+    private final Path rootLocation;
+
+    @Autowired
+    public FileSystemStorageService(StorageProperties properties) {
+        this.rootLocation = Paths.get(properties.getLocation());
+    }
+
+    @Override
+    public Path store(MultipartFile file) {
+        Path archivePath;
+
+        try {
+            if (file.isEmpty()) {
+                throw new StorageException("Failed to store empty file " + file.getOriginalFilename());
```

Comment:
The process doesn't really "fail", but it is not permitted. The message should reflect that.

Does this conditional branch need to be inside the try-catch? The [`isEmpty` method Javadoc](http://docs.spring.io/spring/docs/current/javadoc-api/org/springframework/web/multipart/MultipartFile.html#isEmpty--) says nothing about an IOException being thrown.